src/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -0,0 +1,668 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Represents a JSON array.
+    /// </summary>
+    public partial class JsonArray : JsonNode, IList<JsonNode>, IReadOnlyList<JsonNode>
+    {
+        internal readonly List<JsonNode> _list;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the empty array.
+        /// </summary>
+        public JsonArray() => _list = new List<JsonNode>();
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        public JsonArray(IEnumerable<JsonNode> values) => _list = new List<JsonNode>(values);
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonArray"/> class representing the specified collection of <see cref="string"/>s.
+        /// </summary>
+        /// <param name="values">Collection to represent.</param>
+        /// <exception cref="ArgumentNullException">
+        ///   Some of provided values are null.
+        /// </exception>
+        public JsonArray(IEnumerable<string> values) : this()
+        {
+            foreach (string value in values)
+            {
+                _list.Add(new JsonString(value));
```

Comment:
NIT: I believe @stephentoub already gave you this feedback before, but in here one of the values might be null, which theoretically that's fine but it is weird that if you right today `var str = new JsonString();` that will set the value to string.empty.

Reply:
I think `JsonString` shouldn't accept null, as for such cases we have null `JsonNode`. In here, JsonString constructor will throw an exception. 

Previous comment you mentioned: https://github.com/dotnet/corefx/pull/40107#discussion_r311722058

Reply:
right, but if that's the case, I probably would expect that `var string = new JsonString();` should return me a new JsonString object which has null as value

Reply:
A `null` string in the `values` should produce a JSON null in the output.

Reply:
Ok, I will fix it.